packages/jest-mock/src/index.ts

```diff
@@ -1080,146 +1143,130 @@ export class ModuleMocker {
     }
 
-    return object[methodKey];
-  }
-
-  private _spyOnProperty<T extends object, K extends PropertyLikeKeys<T>>(
-    obj: T,
-    propertyKey: K,
-    accessType: 'get' | 'set',
-  ): Mock<() => T> {
-    if (typeof obj !== 'object' && typeof obj !== 'function') {
-      throw new Error(
-        `Cannot spyOn on a primitive value; ${this._typeOf(obj)} given`,
-      );
+    if (accessType && accessType != 'get' && accessType != 'set') {
+      throw new Error('Invalid accessType supplied');
     }
 
-    if (!obj) {
+    if (typeof object !== 'object' && typeof object !== 'function') {
       throw new Error(
-        `spyOn could not find an object to spy upon for ${String(propertyKey)}`,
+        `Cannot spyOn on a primitive value; ${this._typeOf(object)} given`,
       );
     }
 
-    if (!propertyKey) {
-      throw new Error('No property name supplied');
-    }
-
-    let descriptor = Object.getOwnPropertyDescriptor(obj, propertyKey);
-    let proto = Object.getPrototypeOf(obj);
-
+    let descriptor = Object.getOwnPropertyDescriptor(object, methodKey);
+    let proto = Object.getPrototypeOf(object);
     while (!descriptor && proto !== null) {
-      descriptor = Object.getOwnPropertyDescriptor(proto, propertyKey);
+      descriptor = Object.getOwnPropertyDescriptor(proto, methodKey);
       proto = Object.getPrototypeOf(proto);
     }
-
     if (!descriptor) {
-      throw new Error(`${String(propertyKey)} property does not exist`);
+      throw new Error(`${String(methodKey)} property does not exist`);
     }
-
     if (!descriptor.configurable) {
-      throw new Error(`${String(propertyKey)} is not declared configurable`);
+      throw new Error(`${String(methodKey)} is not declared configurable`);
     }
 
-    if (!descriptor[accessType]) {
-      throw new Error(
-        `Property ${String(
-          propertyKey,
-        )} does not have access type ${accessType}`,
-      );
+    if (this.isMockFunction(descriptor.value)) {
+      return object[methodKey];
+    } else if (accessType == 'get' && this.isMockFunction(descriptor.get)) {
+      return descriptor.get;
+    } else if (accessType == 'set' && this.isMockFunction(descriptor.set)) {
+      return descriptor.set;
     }
 
-    const original = descriptor[accessType];
-
-    if (!this.isMockFunction(original)) {
-      if (typeof original !== 'function') {
+    if (accessType) {
+      if (typeof descriptor[accessType] !== 'function') {
         throw new Error(
-          `Cannot spy the ${String(
-            propertyKey,
-          )} property because it is not a function; ${this._typeOf(
-            original,
-          )} given instead`,
+          `Cannot spy the ${String(accessType)} ${String(
+            methodKey,
+          )} property because it is not a function;
+          ${this._typeOf(descriptor?.[accessType])} given instead`,
         );
       }
+    } else if (typeof descriptor.value !== 'function') {
```

Comment:
@staplespeter What I found by quickly debugging the issue is that, the check we have here is overlooking `descriptor.get` which in our case is a function; `descriptor.value` is `undefined`. I guess we should check both.